### ISSUE TITLE
Error when creating UQFF format model: No field 'count' on type '&metal::QMetalStorage'

### ISSUE THREAD

**Issue by solaoi** [EricLBuehler/mistral.rs#786]
## Describe the bug
I encountered an error while attempting to create a UQFF format model using the following command:

```
cargo run --features metal -- --isq Q4_0 -i plain -m aixsatoshi/Honyaku-13b --write-uqff Honyaku-13b-q4_0.uqff
```

The error message is as follows:

```
error[E0609]: no field `count` on type `&metal::QMetalStorage`
   --> /Users/solaoi/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:211:26
    |
211 |         let size = (self.count * self.dtype.size_in_bytes()) as NSUInteger;
    |                          ^^^^^ unknown field
    |
    = note: available fields are: `dtype`, `device`, `buffer`

error[E0599]: no method named `size_in_bytes` found for enum `quantized::GgmlDType` in the current scope
   --> /Users/solaoi/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:211:45
    |
211 |         let size = (self.count * self.dtype.size_in_bytes()) as NSUInteger;
    |                                             ^^^^^^^^^^^^^ method not found in `GgmlDType`
    |
   ::: /Users/solaoi/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/mod.rs:144:1
    |
144 | pub enum GgmlDType {
    | ------------------ method `size_in_bytes` not found for this enum

error[E0609]: no field `count` on type `&metal::QMetalStorage`
   --> /Users/solaoi/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:223:44
    |
223 |         Ok(read_to_vec::<u8>(&buffer, self.count))
    |                                            ^^^^^ unknown field
    |
    = note: available fields are: `dtype`, `device`, `buffer`
```

## Latest commit or version
master branch（2024/09/22）
[2df0044c8139e4a99b4c9e32a81ea9ce32832599](https://github.com/EricLBuehler/mistral.rs/tree/2df0044c8139e4a99b4c9e32a81ea9ce32832599)

**Comment by ioma8** [comment]
i get the same errors when trying to build the mistral.rs on macos with metal feature:
error[E0609]: no field `count` on type `&metal::QMetalStorage`
   --> /Users/jakubkolcar/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:211:26
    |
211 |         let size = (self.count * self.dtype.size_in_bytes()) as NSUInteger;
    |                          ^^^^^ unknown field
    |
    = note: available fields are: `dtype`, `device`, `buffer`

error[E0599]: no method named `size_in_bytes` found for enum `quantized::GgmlDType` in the current scope
   --> /Users/jakubkolcar/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:211:45
    |
211 |         let size = (self.count * self.dtype.size_in_bytes()) as NSUInteger;
    |                                             ^^^^^^^^^^^^^ method not found in `GgmlDType`
    |
   ::: /Users/jakubkolcar/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/mod.rs:144:1
    |
144 | pub enum GgmlDType {
    | ------------------ method `size_in_bytes` not found for this enum

error[E0609]: no field `count` on type `&metal::QMetalStorage`
   --> /Users/jakubkolcar/.cargo/git/checkouts/candle-c6a149c3b35a488f/8a99f7c/candle-core/src/quantized/metal.rs:223:44
    |
223 |         Ok(read_to_vec::<u8>(&buffer, self.count))
    |                                            ^^^^^ unknown field
    |
    = note: available fields are: `dtype`, `device`, `buffer`

**Comment by EricLBuehler** [comment]
@solaoi @ioma8  I opened #792 which should make some progress to solving the issue, can you please try it out?

**Comment by EricLBuehler** [comment]
@ioma8 @solaoi closing as #792 has been tested and shown to work.

**Comment by solaoi** [comment]
@EricLBuehler 
Thank you for addressing the issue. I've run the command, and it no longer produces any errors, which is great.

I have a follow-up question, though. I was under the impression that executing the command would generate a uqff file. While I understand that it enters the "mistralrs_server::interactive_mode" and I was able to test its functionality, I'm a bit confused about the file output. After exiting with Ctrl+C, I looked for the uqff file, but I couldn't find it anywhere.

Here's the command I tried:

```
cargo run --features metal -- --isq Q4K -i plain -m microsoft/Phi-3.5-mini-instruct --write-uqff phi3.5-mini-instruct-q4k.uqff
```

Could you clarify where I should look for the uqff file, or if there's an additional step I need to take to generate it? Any insights would be greatly appreciated.

**Comment by EricLBuehler** [comment]
Hi @solaoi!

I assume you exited with Ctrl+C after interactive mode starts, which would mean that the UQFF file has been generated. If that is the case, and given the command you ran, the UQFF file should be in the working directory. If not, and interactive mode started, could you please raise a seperate issue?

**Comment by solaoi** [comment]
@EricLBuehler 
Thanks for your response.
I followed the steps exactly as you described, including letting the interactive mode start. However, the UQFF file wasn't created in the current directory after I exited with Ctrl+C.

So I'm raising this issue.
https://github.com/EricLBuehler/mistral.rs/issues/810

The process seems to begin, but the file generation doesn't complete when using the Metal features, unlike with the CPU version where it works fine.
